polaris/yaml.py

```diff
@@ -0,0 +1,401 @@
+            yaml.streams = configs['streams']
+            configs = dict(configs)
+            configs.pop('streams')
+
+        yaml.configs = configs
+        return yaml
+
+    def update(self, configs, quiet=True):
+        """
+        Add config options from a dictionary
+
+        Parameters
+        ----------
+        configs : dict
+            A nested dictionary of config sections, options and values
+
+        quiet : bool, optional
+            Whether or not to print the updated config options as they are
+            replaced
+        """
+        if self.model in configs:
+            # we want one layer deeper
+            configs = configs[self.model]
+        _update_section(configs, self.configs, quiet)
+
+    def write(self, filename):
+        """
+        Write config options to a yaml file
+
+        Parameters
+        ----------
+        filename : str
+            A yaml config file
+        """
+        yaml = YAML(typ='rt')
+        configs = dict(self.configs)
+        if self.streams:
+            configs['streams'] = self.streams
+
+        model_configs = dict()
+        model_configs[self.model] = configs
+
+        with open(filename, 'w') as outfile:
+            yaml.dump(model_configs, outfile)
+
+    def _add_stream(self, stream_name, stream):
+        """
+        Add stream from a dictionary
+        """
+        self.streams[stream_name] = stream
+
+
+def mpas_namelist_and_streams_to_yaml(model, namelist_template=None,
+                                      namelist=None,
+                                      streams=None):
+    """
+    Add config options from a yaml file
+
+    Parameters
+    ----------
+    model : str
+        The name of the model
+
+    namelist_template : str
+        An MPAS namelist template file
+
+    namelist : str, optional
+        An MPAS namelist file
+
+    streams : str, optional
+        An MPAS streams file
+
+    Returns
+    -------
+    yaml : polaris.yaml.PolarisYaml
+        A yaml object with the namelists and streams
+    """
+    yaml = PolarisYaml()
+    yaml.model = model
+    if namelist is not None:
+        yaml.configs = _read_namelist(namelist_template, namelist)
+    if streams is not None:
+        yaml.streams = _streams_xml_to_dict(streams)
+
+    return yaml
+
+
+def main_mpas_to_yaml():
+    parser = argparse.ArgumentParser(
+        description='Convert a namelist and/or streams file to yaml')
+    parser.add_argument("-n", "--namelist", dest="namelist",
+                        required=False,
+                        help="MPAS namelist file")
+    parser.add_argument("-s", "--streams", dest="streams",
+                        required=False,
+                        help="MPAS streams file")
+    parser.add_argument("-t", "--namelist_template", dest="namelist_template",
+                        required=False,
+                        help="MPAS namelist template file (with all namelist "
+                             "options)")
```

Comment:
```suggestion
                             "options. This will typically be ${PATH_TO_MPAS_OCEAN}/default_inputs/namelist.ocean")
```
Maybe add something like this?